src/team_lib/mhba/xccl_mhba_ib.c

```diff
@@ -123,3 +124,213 @@ xccl_status_t xccl_mhba_qp_connect(struct ibv_qp *qp, uint32_t qp_num,
     }
     return XCCL_OK;
 }
+
+xccl_status_t xccl_mhba_init_dc_qps_and_connect(xccl_mhba_team_t *mhba_team, uint32_t *local_data, uint8_t port_num){
+    int i;
+    struct ibv_qp_init_attr_ex attr_ex;
+    struct mlx5dv_qp_init_attr attr_dv;
+    struct ibv_qp_attr qp_attr_to_init;
+    struct ibv_qp_attr qp_attr_to_rtr;
+    struct ibv_qp_attr qp_attr_to_rts;
+    memset(&attr_ex, 0, sizeof(attr_ex));
+    memset(&attr_dv, 0, sizeof(attr_dv));
+    memset(&qp_attr_to_init, 0, sizeof(qp_attr_to_init));
+    memset(&qp_attr_to_rtr, 0, sizeof(qp_attr_to_rtr));
+    memset(&qp_attr_to_rts, 0, sizeof(qp_attr_to_rts));
+
+    attr_ex.qp_type = IBV_QPT_DRIVER;
+    attr_ex.send_cq = mhba_team->net.cq;
+    attr_ex.recv_cq = mhba_team->net.cq;
+    attr_ex.pd = mhba_team->node.shared_pd;
+    attr_ex.cap.max_send_wr = (SQUARED(mhba_team->node.sbgp->group_size / 2) + 1) * MAX_OUTSTANDING_OPS *
```

Comment:
is this computed based on the minimal block_size = 2 ?

Reply:
yes, it's just until will have that polling mechanism. soon, it will come together with the transition to memic